src/main/java/com/pagsestagio/movieapi/controller/FilmeController.java

```diff
@@ -24,53 +24,53 @@ public FilmeController(FilmeService filmeService){
         this.filmeService = filmeService;
     }
 
-    @GetMapping("/{id}")
-    public ResponseEntity<FilmeResposta> pegarFilmePeloId(@PathVariable Integer id){
-        FilmeResultadoRetornaFilmeOuExcecao retornoService = filmeService.pegarFilmePeloId(id);
+    @GetMapping("/{idpublico}")
```

Comment:
não havíamos conversado de você manter esses endpoints iguais, marcar eles como deprecated.
e cria uma nova versão?

Tipo no sentindo de evoluir sem quebrar ao compatibilidade.
imagina só. você até deletar esses endpoints depois.
Mas tem que dar tempo para os clientes que as atualizar.
ai depois você faz o teste do grito.

Reply:
Isso é uma boa questão, além do exemplo que o Manduzinho deu, podemos trazer um exemplo da própria pags.
Imagina que a API da pags de pamento online for alterada bruscamente dessa maneira? Atualizando valores de parametros obrigatórios ou algo do tipo, isso iria gerar uma quebra gigantesca de contrato com nossos clientes e te garanto que geraria várias crises por aí hehe.

Claro que tbm entendo que os clientes da API devem se preocupar com essa questão e devem se preparar criando [ACL](https://medium.com/solutions-architecture-patterns/anti-corruption-layer-pattern-bd75e1f2be7f) para essa comunicação, mas de qualquer maneira, a partir do momento que você muda nome dos parametros obrigatórios isso iria gerar bastante problemas de qualquer maneira para o cliente.

O ideal mesmo é alterar o que é preciso em outra versão do endpoint e colocar uma data limite para o suporte do primeiro endpoint.

Reply:
Algo que não entendo é tipo... vamo pegar como exemplo a classe Filme, eu ela foi praticamente toda remodelada, eu teria que ter uma classe Filme que é a da versão atual e uma FilmeVersaoAntiga por exemplo, que funcione do modo antigo? E isso pra todas as classes que foram alteradas desse jeito? Ter essas duas versões?

Reply:
Outro ponto, eu continuaria a usar a estrutura de dados Map pra essa "versão antiga"? Ou a ideia é justamente usar o banco de dados h2 mas tudo que a requisição receber e retornar tem que permanecer o mesmo de antes, por assim dizer?

Reply:
Então como seu código lida com as mudanças internamentes deve ser transparente para o cliente da api.
você pode até ter trocado seu código para o python. 

para quem chama seu endpoint não muda nada.
como você resolve isso no código? ai depende de você. 
ter classes versionadas.
traduzir no controller. 

Sim.
talvez sua tabela tenha que capacidade de sustentar a classe antiga.
algo como: idIntLegacy.

ai sua classe teria algo como id
id
idPublico
idIntLegacy
nome

ai esse idIntLegacy você só manteria esse para retrocompatibilidade.
e só quem subiu usando o outro caminho veria isso.

o cara quer um map.
então devolva um map.
Talvez o map com só os que tem idLegado.

tipo, garante que não quebrou nada.
mas o cara quer todos os filmes, não só os legados? vai ter que ir na api nova